workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -121,11 +121,12 @@ def restore(self):
 
     def _restore_deployment_envs(self):
         deps = {}
-        for tenant in self._client.tenants.list():
+        tenants = [get_tenant_name()] if self._snapshot_version < V_4_0_0 \
```

Comment:
For readability perhaps have this line set using a standard if statement.. you can do this after this PR is merged.